salt/states/user.py

```diff
@@ -438,7 +438,24 @@ def present(name,
     # hash_password is True, then hash it.
     if password and hash_password:
         log.debug('Hashing a clear text password')
-        password = __salt__['shadow.gen_password'](password)
+        # in case a password is already set, it will contain a Salt
+        # which should be re-used to generate the new hash, other-
+        # wise the Salt will be generated randomly, causing the
+        # hash to change each time and thereby making the
+        # user.present state non-idempotent.
+        algorithms = {
+            '1':  'md5',
```

Comment:
But this is insecure. I think we should either kill this option or at least scream all around the place in logs if anyone will use this.

Reply:
I do agree that "screaming all around the place in logs" is the most reasonable reaction here…

Added a corresponding warning.